spock-core/core.gradle

```diff
@@ -49,20 +47,21 @@ jar {
       'Implementation-Vendor': 'spockframework.org',
       'Automatic-Module-Name': 'org.spockframework.core'
     )
-    instruction 'Export-Package', 'org.spockframework.*', 'spock.*'
-    instruction 'Embed-Dependency', 'groovy;inline=false', 'junit;inline=false'
-    instruction 'Import-Package',
-        'groovy.*;version="[2.4,4)"',
-        'org.codehaus.groovy.*;version="[2.4,4)"',
-        'org.junit.*;version="[4,6)"',
-        'org.hamcrest.*;version="[1,2)";resolution:=optional',
-        'org.objenesis.*;version="[2,3)";resolution:=optional',
-        'org.apache.tools.ant.*;version="[1.7,2)";resolution:=optional',
-        'net.bytebuddy.*;version="[1.9,2)";resolution:=optional',
-        'net.sf.cglib.*;version="[3.2,4)";resolution:=optional',
-        'org.objectweb.asm.*;version="[5.2,6)";resolution:=optional',
-        'org.w3c.dom.*'
   }
+  bnd(
+    'Export-Package': ['org.spockframework.*', 'spock.*'].join(','),
+    'Embed-Dependency': ['groovy;inline=false', 'junit;inline=false'].join(','),
+    'Import-Package': [
+      /groovy.*;version="[${groovyVersion},${org.gradle.util.VersionNumber.parse(groovyVersion).major + 1})"/,
+      /org.codehaus.groovy.*;version="[${groovyVersion},${org.gradle.util.VersionNumber.parse(groovyVersion).major + 1})"/,
+      /org.junit.*;version="[${libs.junitBom.version},${org.gradle.util.VersionNumber.parse(libs.junitBom.version).major + 1})"/,
+      /org.hamcrest.*;version="[${libs.hamcrest.version},${org.gradle.util.VersionNumber.parse(libs.hamcrest.version).major + 1})";resolution:=optional/,
+      /org.objenesis.*;version="[${libs.objenesis.version},${org.gradle.util.VersionNumber.parse(libs.objenesis.version).major + 1})";resolution:=optional/,
+      /net.bytebuddy.*;version="[${libs.bytebuddy.version},${org.gradle.util.VersionNumber.parse(libs.bytebuddy.version).major + 1})";resolution:=optional/,
+      /net.sf.cglib.*;version="[${libs.cglib.version},${org.gradle.util.VersionNumber.parse(libs.cglib.version).major + 1})";resolution:=optional/,
+      /org.objectweb.asm.*;version="[${libs.asm.version},${org.gradle.util.VersionNumber.parse(libs.asm.version).major + 1})";resolution:=optional/,
```

Comment:
I hope that you ran it locally and it generated sensible entries? :-)

Reply:
I ran it, and it looked ok, but I don't really know that much about osgi, so maybe someone else can take a look.

Reply:
@rotty3000 You added OSGI metadata in JUnit 5 (https://github.com/junit-team/junit5/pull/2095) and it seems to have much greater knowledge about that than we have :-)

Could you take a look at that PR and provide some comments/suggestions?